lizmap/modules/lizmap/lib/Project/QgisProject.php

```diff
@@ -1464,25 +1464,43 @@ protected function getFieldConfiguration($layerXml)
         return $edittypes;
     }
 
+    protected function getTabFromAttributes($attributes)
+    {
+        $tab = array();
+
+        foreach ($attributes as $key => $value) {
+            $tab[(string) $key] = (string) $value;
+        }
+
+        return $tab;
+    }
+
     protected function getEditType($layerXml)
     {
-        $edittypes = $layerXml->edittypes;
+        $edittypes = $layerXml->xpath('.//edittypes');
```

Comment:
add a comment:

```
// <edittypes> can be a child of <maplayer>, but can be also elsewhere inside a <maplayer><map-layer-style-manager><map-layer-style><qgis/></map-layer-style></map-layer-style-manager></maplayer>
```

Reply:
`getEditType()` is called after already retrieving the edittype elements in `readFormControls()`. So, to avoid to call twice `xpath` for the same thing, `getEditType` should be called with the `<edittypes>` element, no the `<maplayer>` element.

Reply:
so, add the comment into `readFormControls()` instead.